src/lib/tabs/tab-group.ts

```diff
@@ -290,4 +291,12 @@ export class MatTabGroup extends _MatTabGroupMixinBase implements AfterContentIn
     this._tabBodyWrapperHeight = this._tabBodyWrapper.nativeElement.clientHeight;
     this._renderer.setStyle(this._tabBodyWrapper.nativeElement, 'height', '');
   }
+
+  /** Handle click events, setting new selected index if appropriate. */
+  _handleClick(tab: MatTab, tabHeader: MatTabHeader, idx: number) {
+    if (!tab.disabled) {
+      tabHeader.focusIndex = idx;
+      this.selectedIndex = idx;
+    }
```

Comment:
Optionally this could be reduced to: `this.selectedIndex = tabHeader.focusIndex = idx`.